### DISCUSSION TITLE
TypeError: Cannot read property 'removeChild' of null

### DISCUSSION THREAD

**Discussion by Saifrazaa** [vercel/next.js#24466] [Help]
![image](https://user-images.githubusercontent.com/28540950/116074198-d568d300-a6aa-11eb-82d1-1fe488a9c818.png)
i am facing this issue in head-manager file. can anyone tell me what is the issue and any solution for that?

**Comment by bobarros** [comment]
Too little info. 
But it seems to me that oldTags it's an HTML collection, right? You have this collection by doing something like:

```javascript 
const h1All = d.getElementsByTagName('h1');
```

In this case, forEach does not work. You need something like:

```javascript 
    for (let i = 0; i < h1All.length; i++) {
      const element = h1All[i];
      element.style.backgroundColor = 'black';
    }
```

When asking for help, you should make an effort to give all the information you have on the issue ;)

**Reply by Saifrazaa** [reply]
@bobarros this is my first time i am asking any question on community platform. The information i would like to provide about this error is:
I am using React.js with Next.js

React.js version:"16.13.1",
Next.js version:"9.5.5"

having this issue in head-manager.ts file
is this info enough?


> Too little info.
> But it seems to me that oldTags it's an HTML collection, right? You have this collection by doing something like:
> 
> ```js
> const h1All = d.getElementsByTagName('h1');
> ```
> 
> In this case, forEach does not work. You need something like:
> 
> ```js
>     for (let i = 0; i < h1All.length; i++) {
>       const element = h1All[i];
>       element.style.backgroundColor = 'black';
>     }
> ```
> 
> When asking for help, you should make an effort to give all the information you have on the issue ;)

**Reply by bobarros** [reply]
Did you try the solution that I offered? Not using foreach.
It's not only about versions. We have no idea what oldTags and oldTag are in your code.
As far as I know from what you're showing, they could be strings 🤷

Just look that difference between you post and this one --> https://github.com/vercel/next.js/discussions/24382

Whenever you try to do something for the first time, take a look at how other people are doing it. Might be a great start 👍

**Reply by Saifrazaa** [reply]
because of your solution i got the thing which is making problem and it is "var f=d.getElementsByTagName(s)[0]". whenever i am commenting the script tag having this line the error disappears.

Thanks man for the help . really appreciated !!!

**Comment by bobarros** [comment]
You really don't know what d represents in my code?
You should use document.getElementsBy... instead.

And if you don't know what document is, it seems that you should take some steps back and learn some dom and js fundamentals, since you're trying to append and remove children elements already.

**Reply by Saifrazaa** [reply]
i know d in your code and i also know document . 
i found the issue and it's solution thanks.

**Reply by bobarros** [reply]
Last hint. To help others, describe the real problem and your solution. It might help someone else in the future ;)
